src/Microsoft.AspNetCore.SignalR.Core/HubOptions`T.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNetCore.SignalR
+{
+    public class HubOptions<THub> : HubOptions where THub : Hub
+    {
+        public new TimeSpan? NegotiateTimeout { get; set; } = null;
```

Comment:
It feels really weird to me that these properties are defined again and are using new to hide the original properties. Is it because you want NegotiateTimeout and KeepAliveInterval to be nullable here? What scenario is this for?

Reply:
Yea, maybe they should just be null in the base class.

Reply:
Good question. 
The properties are null in this class and not the base class because we want the default values from the base class, but we need to be able to tell if a specific `HubOptions<THub>` has been configured (aka the options are not null) 
It allows us to differentiate between the scenarios in which there are only global options defined, global and local so we know which settings to enforce. 

That being said, yeah we could just make them null in the base class but I do like the idea of default values.

Reply:
We can just have defaults come from the IConfigureOptions<HubOptions> instead of being on the HubOptions class. It's a bit cleaner.